qi-lib/flow/core/compiler/deforest/cps.rkt

```diff
@@ -23,6 +23,13 @@
     [(_ f) f]
     [(_ [op (f ...)] rest ...) (op f ... (inline-compose1 rest ...))]))
 
+(define-syntax inline-consing
+  (syntax-rules ()
+    [(_ state () rest ...) (inline-consing state rest ...)]
+    [(_ state (arg) rest ...) (inline-consing (cons arg state) rest ...)]
+    [(_ state) state]
+    ))
```

Comment:
Some examples of how this works would be helpful; it looks like a macro that
reverses `(list state rest ...)` (where each `rest` can be nothing or an `arg`)?
(This is addressed later and just needs added with this commit (1a79dc04d21baab790c2fe6fa3a804ab0b376e54) instead.)

Reply:
Addressed by rebases.